Add test cases of reusable block

fluiddot opened the pull request:
The Reusable block will be released on version `1.54.0` so I've added the test cases that we should cover on the sanity checks.

fluiddot commented on test-cases/README.md:
```diff
@@ -20,6 +20,8 @@ Gutenberg:
 - [Verse block](./gutenberg/verse.md)
 - [Video block](./gutenberg/video.md)
 - [File block](./gutenberg/file.md)
+- [Audio block](./gutenberg/audio.md)
```
@jd-alexander I realized that the Audio block wasn't included in this index so went ahead and added it, let me know if it's ok.

jd-alexander replied:
@fluiddot Thanks for adding it here. 👍🏾

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+
+**For testing this block, it's required to create reusable blocks via the web version, here are the steps:**
+
+- Navigate to reusable blocks page (URL: `https://wordpress.com/types/wp_block/<YOUR_WPCOM_SITE_DOMAIN>`).
```
I think that sometimes WPCOM will ask you to select the site anyway, like this:

![Screenshot 2021-05-27 at 20 41 06](https://user-images.githubusercontent.com/1032913/119872243-ec117c80-bf2b-11eb-8adf-97824b24ba7c.png)

So, what do you think adding a note about that in the instruction?

fluiddot replied:
Oh good point, I'll update the step then with:

- Navigate to reusable blocks page `https://wordpress.com/types/wp_block/`.
- Select the WP.com site for testing.

hypest replied:
I see this got addressed with https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R78-R79 👍 , thanks!

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+
+**Expected Behavior**
+- Expect to see the reusable block and its content.
+- Epxect to see the block's name in the block's header.
```
```suggestion
- Expect to see the block's name in the block's header.
```

fluiddot replied:
Thanks for the typo correction 🙇 .

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+- Tap on ➕ button.
+- Tap on "Reusable" tab.
+- Tap on a reusable block.
+- Once inserted, tap on the block.
```
Wdyt about adding a "to edit it" at the end of the sentence? Looks like that's how describe the goal [later in the text](https://github.com/wordpress-mobile/test-cases/pull/79/files#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R33).

fluiddot replied:
Yeah, it will be clearer to add that part as the action is meant to edit the block.

hypest replied:
I see this got addressed with https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R25, thanks!

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+
+**Deleted reusable block**
+
+**Steps**
```
To be honest, I wasn't sure if all the steps were meant to be performed on the web or not. I figured it out and the test works OK, and looks like only the last step (`add the first reusable block to a post/page`) is meant to be performed on the mobile so, wdyt about making it more clear? For example, we could have something like this:

```
-> On the web (ref/link)
  - Create two reusable blocks.
  - Add the second reusable block as part of the content of the first one and update it.
  - Delete the second reusable block from the "Reusable blocks page" (referenced above).
  - The second reusable block will go to the trashed tab, delete it permanently.
-> Switch to mobile device for the last step
  - Add the first reusable block to a post/page.
```

I know that the last step is not mandatory to be performed on mobile, but since we don't have an explicit step for the tester to switch to mobile, we might as well request that "add block..." step be done on mobile, wdyt?

fluiddot replied:
I totally agree, I'll add the extra steps to make it clearer in which platforms the actions should be performed.

hypest replied:
Thanks for the updates (I see the "switch to mobile" instruction [here for example](https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R84)). I could still see some usefulness on mentioning on the first step which platform the tester should be, to make it easier to remember when scanning through the steps? I [left a comment on that directly on the commit here](https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#r642420368).

fluiddot replied:
Agreed, I've just committed a change regarding this in this [commit](https://github.com/wordpress-mobile/test-cases/pull/79/commits/34b73333100a1c0a8c319d7860230361f5f1a85f).

hypest replied:
Resolving as it [got addressed here](https://github.com/wordpress-mobile/test-cases/pull/79#discussion_r642472744).

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+**Steps**
+
+- Create a reusable block.
+- Edit the reusable block, add the same reusable block as part of the content and update it.
```
Wdyt about adding a note here that Gutenberg web will show a `Block cannot be rendered inside itself.` to make it expected?

fluiddot replied:
Sure! In fact, we expect the same behavior on both platforms Web and Mobile.

hypest replied:
I see this addressed in https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R104, thanks!

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -0,0 +1,104 @@
+
+- Create a reusable block.
+- Edit the reusable block, add the same reusable block as part of the content and update it.
+- Add the reusable block to a post/page.
```
Wdyt about adding a " ... on mobile" at the end to make it clear where to perform it?

hypest replied:
I see this addressed in various places with https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16#diff-6c976df25a3a675826c7dcc37f0922bd611004286dcba71e631192648fdd84c5R105, thanks!

hypest reviewed (commented):
Looking good @fluiddot ! I've left a few comments, not really major ones.

fluiddot commented:
> Looking good @fluiddot ! I've left a few comments, not really major ones.

Thank you very much @hypest for the feedback ❤️ ! I've just pushed [this commit](https://github.com/wordpress-mobile/test-cases/pull/79/commits/d9411dd6cde52786887535b7104a17a6c3978d16) that addresses the comments you posted, I'd really appreciate if you could review them, thanks 🙇 !

hypest commented on test-cases/gutenberg/reusable.md:
```diff
@@ -74,10 +75,13 @@ The reusable blocks page can be also used for editing and deleting blocks.
 
 **Steps**
-- Create two reusable blocks.
+- Navigate to the [reusable blocks page](https://wordpress.com/types/wp_block).
```
Wdyt about a quick nudge to which platform this part of the steps should be performed on? Something like:
```suggestion
- On a web browser, navigate to the [reusable blocks page](https://wordpress.com/types/wp_block).
```

fluiddot replied:
Sure! I'll update this part in both tests `TC003` and `TC004`.

fluiddot replied:
I've just changed the steps that referenced the navigation to the reusable blocks page in this [commit](https://github.com/wordpress-mobile/test-cases/pull/79/commits/34b73333100a1c0a8c319d7860230361f5f1a85f).

hypest reviewed (approved):
LGTM!
